File: P03_CRUD/P03_CRUD/src/main/java/com/cims/crud/entities/Material.java
package com.cims.crud.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name="material")
public class Material {
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="m_id")
	int m_id;
	
	@Column(name="m_name")
	String m_name;
	
	@ManyToOne
	@JoinColumn(name="cat_id")//foreign key category
	Category cat_id;
	
	@ManyToOne 
	@JoinColumn(name="unit_id")//foreign key unit
	Unit unit_id;
	
	@Column(name="description")
	String description;

}

File: CIMS/cims/Models/Material.cs
using System;
using System.Collections.Generic;

namespace cims.Models;

public partial class Material
{
    public int MId { get; set; }

    public string MName { get; set; } = null!;

    public int CatId { get; set; }

    public int UnitId { get; set; }

    public string? Description { get; set; }

    public virtual Category Cat { get; set; } = null!;

    public virtual ICollection<MaterialProject> MaterialProjects { get; set; } = new List<MaterialProject>();

    public virtual ICollection<MaterialRequest> MaterialRequests { get; set; } = new List<MaterialRequest>();

    public virtual Unit Unit { get; set; } = null!;
}
